src/cocotb/task.py

```diff
@@ -210,21 +212,18 @@ def cancel(self, msg: Optional[str] = None) -> None:
         if self.done():
             return
 
-        self._cancelled_error = CancelledError(msg)
-        warnings.warn(
-            "Calling this method will cause a CancelledError to be thrown in the "
-            "Task sometime in the future.",
-            FutureWarning,
-            stacklevel=2,
-        )
+        if msg is not None:
+            self._cancelled_error = CancelledError(msg)
+        else:
+            self._cancelled_error = CancelledError()
+
+        # throw CancelledError and unschedule
+        self._advance(Error(self._cancelled_error))
```

Comment:
If `cancel()` is called before this finishes we'll have problems, but the followup should fix that.